app/views/entities/show.rb

```diff
@@ -89,23 +90,45 @@ def thumbnail
       end
     end
 
+    def unreferenced?
+      !referenced_records.detect { |_key, records| records[:total][:value] != 0 }
+    end
+
+    def referenced_records
+      @referenced_records ||= {}
+      @entity.search_keys.each do |key|
+        @referenced_records[key] ||= begin
+          return { search_results: [], total: { value: 0, formatted: '0' } } unless @entity.respond_to?(:search_query)
+          @response = @search_results[key]
```

Comment:
Change the use of `@response` in this method to just `response` and pass that to `#document_presenter`.